src/main/java/com/exasol/adapter/RequestDispatcher.java

```diff
@@ -39,54 +29,39 @@ public static synchronized RequestDispatcher getInstance() {
      * @throws AdapterException in case the request type is not recognized
      */
     public static String adapterCall(final ExaMetadata metadata, final String rawRequest) throws AdapterException {
-        return getInstance().executeAdapterCall(metadata, rawRequest);
-    }
-
-    @SuppressWarnings("squid:S2139")
-    private String executeAdapterCall(final ExaMetadata metadata, final String rawRequest) throws AdapterException {
         try {
-            logVersionInformation();
-            logRawRequest(rawRequest);
-            final AdapterRequest request = new RequestParser().parse(rawRequest);
-            configureAdapterLoggingAccordingToRequestSettings(request);
-            final VirtualSchemaAdapter adapter = findResponsibleAdapter(request);
-            return processRequest(request, adapter, metadata);
+            return processAdapterCall(metadata, rawRequest);
         } catch (final Exception exception) {
             LOGGER.severe(exception::getMessage);
             LOGGER.log(Level.FINE, "Stack trace:", exception);
             throw exception;
         }
     }
 
-    private VirtualSchemaAdapter findResponsibleAdapter(final AdapterRequest request) {
-        final String name = request.getAdapterName();
-        return AdapterRegistry.getInstance().getAdapterForName(name);
+    private static String processAdapterCall(final ExaMetadata metadata, final String rawRequest)
+            throws AdapterException {
+        logVersionInformation();
+        logRawRequest(rawRequest);
+        final AdapterRequest adapterRequest = parseRequest(rawRequest);
+        configureAdapterLoggingAccordingToRequestSettings(adapterRequest);
+        final AdapterCallExecutor adapterCallExecutor = getAdapterCallExecutor();
```

Comment:
Is it okay that we create it from a scratch each time?

Reply:
If we create a new AdapterCallExecutor each time we could send adapterRequest and metadata as the constructor parameters

Reply:
In the previous way we were also creating it from scratch every time. The difference is the now we have decoupled the responsibilities of building and execution, which before was all in the same class.